### PR TITLE
leverage ITuple interface

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by SimonCropp** [comment]
merge conflict resolved

File: test/Serilog.Tests/Capturing/PropertyValueConverterTests.cs [hunk 1]
```diff
@@ -396,13 +396,17 @@ public void AllTupleLengthsUpToSevenAreSupportedForCapturing()
             Assert.IsType<SequenceValue>(_converter.CreatePropertyValue(t));
     }
 
+#if !IVALUETUPLE
+
     [Fact]
     public void EightPlusValueTupleElementsAreIgnoredByCapturing()
```
Comment (SimonCropp) [inline test/Serilog.Tests/Capturing/PropertyValueConverterTests.cs:402]: @nblumhardt 

i wasnt sure if 8 was a by design choice? or u just didnt want to bloat the conversion code with too many tuple conversions.

if it is by design, let me know and i will make the ITuple respect this

**Comment by SimonCropp** [comment]
@nblumhardt thoughts on this on?

File: test/Serilog.Tests/Capturing/PropertyValueConverterTests.cs (hunk 1 above)
Comment (nblumhardt) [reply test/Serilog.Tests/Capturing/PropertyValueConverterTests.cs:402]: Not by design - I think dropping this limitation is good :+1:

**Comment by nblumhardt** [comment]
LGTM!

**Pull request merged**